app/src/main/java/com/example/tasky/feature_agenda/presentation/Agenda.kt

```diff
@@ -0,0 +1,115 @@
+import androidx.compose.material3.Card
+import androidx.compose.material3.CardDefaults
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.navigation.NavController
+import com.example.tasky.AuthNavRoute
+import com.example.tasky.R
+import com.example.tasky.common.domain.error.DataError
+import com.example.tasky.common.presentation.CreateErrorAlertDialog
+import com.example.tasky.common.presentation.LoadingSpinner
+
+@Composable
+fun AgendaContent(
+    navController: NavController,
+    agendaViewModel: AgendaViewModel = hiltViewModel()
+) {
+    val uiState by agendaViewModel.uiState.collectAsStateWithLifecycle()
+    val viewState by agendaViewModel.viewState.collectAsStateWithLifecycle()
+
+    Box(
+        modifier = Modifier
+            .fillMaxSize()
+            .background(Color.Black)
+            .safeDrawingPadding()
+    ) {
+        AgendaToolbar(
+            initials = agendaViewModel.getInitials(),
+            onProfileClick = { agendaViewModel.logOutClicked() }
+        )
+
+        Card(
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(top = 75.dp),
+            shape = RoundedCornerShape(
+                topStart = 30.dp, topEnd = 30.dp
+            ),
+            colors = CardDefaults.cardColors(
+                containerColor = Color.White
+            )
+        ) {
+            Column(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .padding(16.dp)
+                    .verticalScroll(rememberScrollState()),
+                horizontalAlignment = Alignment.CenterHorizontally
+            ) {
+                Text("Top Text")
+            }
+        }
+
+        when (viewState) {
+            is AgendaViewState.Loading -> {
+                // Show a loading indicator
+                LoadingSpinner()
+            }
+
+            is AgendaViewState.Success -> {
+                // Handle success by navigating to LoginScreen
+                navController.navigate(AuthNavRoute)
+            }
+
+            is AgendaViewState.Failure -> {
+                // Show an Alert Dialog with API failure Error code/message
+                val message =
+                    (viewState as AgendaViewState.Failure).dataError.toLogOutErrorMessage(
+                        context = LocalContext.current
+                    )
+                LaunchedEffect(message) {
+                    agendaViewModel.onShowErrorDialog(message)
+                }
+            }
```

Comment:
Interesting concept, how do you distinguish between different types of success? What speaks against integrating this in your UI state so you don't need multiple states per screen? You could just make this extra fields like there.

Reply:
looking back at this, I'm not sure why I made those states in ViewStates. ViewStates should reflect specific UI states. I went back and changed some states. Yes we only need one UI state and can use extra fields. what was i thinking..